test/lib/eslint/BaseEslintEngine.test.ts

```diff
@@ -0,0 +1,150 @@
+import { BaseEslintEngine, StaticDependencies, EslintStrategy } from "../../../src/lib/eslint/BaseEslintEngine";
+import { Rule, RuleGroup, RuleTarget, /*ESRule, ESReport*/ } from '../../../src/types';
+import { expect } from 'chai';
+// import { CLIEngine } from 'eslint';
+import Mockito = require('ts-mockito');
```

Comment:
Switched to using Mockito from Sinon since this adapts much better to the Test-harness type of testing we need here.